src/ModelContextProtocol/Protocol/Transport/SseClientSessionTransport.cs

```diff
@@ -294,13 +294,16 @@ private void HandleEndpointEvent(string data)
             else
             {
                 // If the endpoint is a relative URI, we need to combine it with the relative path of the SSE endpoint
-                var hostUrl = _sseEndpoint.AbsoluteUri;
-                if (hostUrl.EndsWith("/sse", StringComparison.Ordinal))
-                    hostUrl = hostUrl[..^4];
+                var baseUriBuilder = new UriBuilder(_sseEndpoint);
 
-                var endpointUri = $"{hostUrl.TrimEnd('/')}/{data.TrimStart('/')}";
+                // Check that paths end in "/sse" and remove if needed
+                if (baseUriBuilder.Path.EndsWith("/sse", StringComparison.Ordinal))
+                {
+                    baseUriBuilder.Path = baseUriBuilder.Path[..^4];
+                }
```

Comment:
```suggestion
```

I know the "/sse" isn't new, but it should no longer be necessary now that you're using UriBuilder to properly handle relative URIs.